### PR TITLE
[CARBONDATA-2434] Add ExternalTableExample and LuceneDataMapExample

### PR DESCRIPTION
**Pull request by chenliang613**
For preparing 1.4.0 release.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [X] Any interfaces changed?
 NA
 - [X] Any backward compatibility impacted?
 NA
 - [X] Document update required?
NA
 - [X] Testing done
DONE
       
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5627/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4467/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5674/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4514/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4751/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4517/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5677/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4754/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/LuceneDataMapExample.scala [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+         | SELECT id,name
+         | FROM personTable where TEXT_MATCH('id:test1 -id:good1')
+      """.stripMargin).show(100)
+
+    // 3. TEXT_MATCH_WITH_LIMIT usage:
+//    spark.sql(
+//      s"""
+//         | SELECT id,name
+//         | FROM personTable where TEXT_MATCH_WITH_LIMIT('id:test1,10')
+//      """.stripMargin).show()
```
Comment (ravipesala) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/LuceneDataMapExample.scala]: Uncomment it and change to `TEXT_MATCH_WITH_LIMIT('id:test1',10)` to work

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5737/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4579/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**